lib/logflare/backends/ingest_event_queue/broadcast_worker.ex

```diff
@@ -0,0 +1,77 @@
+defmodule Logflare.Backends.IngestEventQueue.BroadcastWorker do
+  @moduledoc """
+  A worker that broadcasts all soruce-backend buffer statistics periodically for the entire node.
+  """
+  use GenServer
+  alias Logflare.Source
+  alias Logflare.Backends.IngestEventQueue
+  alias Logflare.PubSubRates
+  require Logger
+  @ets_table_mapper :ingest_event_queue_mapping
+
+  @interval 1_500
+
+  def start_link(opts) do
+    GenServer.start_link(__MODULE__, opts)
+  end
+
+  def init(opts) do
+    state = %{interval: Keyword.get(opts, :interval, @interval)}
+    Process.send_after(self(), :broadcast, state.interval)
+    {:ok, state}
+  end
+
+  def handle_info(:broadcast, state) do
+    traverse_and_broadcast()
+    Process.send_after(self(), :broadcast, state.interval)
+    {:noreply, state}
+  end
+
+  defp traverse_and_broadcast do
+    case :ets.first(@ets_table_mapper) do
+      :"$end_of_table" ->
+        :ok
+
+      sid_bid ->
+        do_broadcast(sid_bid)
+        next_and_broadcast(sid_bid)
+    end
+  end
+
+  defp next_and_broadcast(prev_key) do
+    case :ets.next(@ets_table_mapper, prev_key) do
+      :"$end_of_table" ->
+        :ok
+
+      sid_bid ->
+        do_broadcast(sid_bid)
+        next_and_broadcast(sid_bid)
+    end
+  end
```

Comment:
just curious, why not `ets:foldl`?

Reply:
yea possible, `select/3` would be [faster](https://www.erlang.org/doc/apps/stdlib/ets.html#select/3) but foldl would be simpler to read.

Reply:
done